hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactingMemStore.java

```diff
@@ -448,17 +453,43 @@ private void releaseResources() {
     * There is at most one thread per memstore instance.
     */
     private void doCompact() {
-
-      ImmutableSegment result = SegmentFactory.instance()  // create the scanner
-          .createImmutableSegment(getConfiguration(), getComparator(),
-              CompactingMemStore.DEEP_OVERHEAD_PER_PIPELINE_ITEM);
-
-      // the compaction processing
-      try {
-        // Phase I: create the compacted MutableCellSetSegment
-        compactSegments(result);
-
-        // Phase II: swap the old compaction pipeline
+      int cellsAfterComp = versionedList.getNumOfCells();
```

Comment:
This code was actually tested. For now the code is not invoked because COMPACTION_PRE_CHECK is false. Why do you think this code should be removed?